### PR TITLE
Download reporting

### PR DESCRIPTION
**Pull request by mikejolley**
This is a continuation of https://github.com/woocommerce/woocommerce/pull/16418

TODO:

- [x] PHPCS

### CODE REVIEW

**Comment by claudiulodro** [comment]
Deprecation notices are breaking a bunch of the tests: https://travis-ci.org/woocommerce/woocommerce/jobs/301563517

File: includes/class-wc-install.php [hunk 1]
```diff
@@ -560,7 +560,7 @@ private static function get_schema() {
 ) $collate;
 CREATE TABLE {$wpdb->prefix}woocommerce_downloadable_product_permissions (
   permission_id BIGINT UNSIGNED NOT NULL auto_increment,
-  download_id varchar(32) NOT NULL,
+  download_id varchar(36) NOT NULL,
```
Comment (claudiulodro) [inline includes/class-wc-install.php]: Do we need an update script for this, too? Otherwise old DBs won't properly handle the new download IDs?

**Comment by claudiulodro** [comment]
Looks good and works well. Just a question and the tests need fixing.

File: includes/class-wc-install.php (hunk 1 above)
Comment (mikejolley) [reply includes/class-wc-install.php]: dbdelta handles it

**Pull request merged**